### ISSUE TITLE
0.4.2 Headings not properly rendered when newlines are introduced

### ISSUE THREAD

**Issue by grenadier117** [flutter/flutter_markdown#249]
H1, H2, etc. are not parsed correctly and show "# text" when new lines are added to the data.
I can't reproduce this issue in 0.4.1

Reproducible build code:

```dart
  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: Text(widget.title),
      ),
      body: MarkdownBody(
        data:
          '# H1  \n'
          '## H2  \n'
          '~~strikethrough~~  \n'
          '*italics*  \n'
          '**bold**'
      )
    );
  }
```

Output:

![image](https://user-images.githubusercontent.com/16449505/89070844-7a2bcf00-d32a-11ea-90ac-495afeac23ea.png)

**Comment by DaveWoodCom** [comment]
I can confirm the same (sort of) issue.

`data = "_This is an italized title_  \r\n  \r\n* **Point:** Something interesting.  \r\n* **Another Point:** Less interesting  \r\n* **One more:** Not really interesting at all  \r\n"`

Screenshots rendered with 0.4.1 vs 0.4.2/0.4.3

![0 4 1](https://user-images.githubusercontent.com/1012467/90821736-652ad600-e301-11ea-9e0f-5d42cf9a2275.png)
![0 4 2](https://user-images.githubusercontent.com/1012467/90821754-6bb94d80-e301-11ea-9170-a7eb51f79c58.png)

Note though that it doesn't appear to be a new line issue, but a change in how `[space][space][newline]` is handled.

Changing to:

`data = "_This is an italized title_\r\n\r\n* **Point:** Something interesting.\r\n* **Another Point:** Less interesting\r\n* **One more:** Not really interesting at all\r\n"`

Renders correctly again.

**Comment by mjordan56** [comment]
@grenadier117 and @DaveWoodCom Once the changes from PR #264 get pushed to pub.dev in a build this should resolve your issue. The changes in PR #264 revert the newline behavior back to the behavior in version 0.4.2 of the flutter_markdown package.

@domesticmouse this issue can be closed.

**Comment by domesticmouse** [comment]
PR #264 has been released as part of 0.4.4.
